staging/src/k8s.io/apiserver/pkg/util/peerproxy/peerproxy_handler_test.go

```diff
@@ -192,6 +192,23 @@ func TestPeerProxy(t *testing.T) {
 			apiserver_rerouted_request_total{code="503"} 2
 			`,
 		},
+		{
+			desc:                 "503 if one apiserver's endpoint lease wasnt found but another valid (unreachable) apiserver was found",
```

Comment:
Do we also need a new test for the hostinfo case?

Reply:
Added a test case for when all peers have invalid hostinfo. Is this what you meant?